contracts/base/LockAndMsgSender.sol

```diff
@@ -1,35 +1,27 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity ^0.8.24;
 
-import {Constants} from '../libraries/Constants.sol';
+import {Locker} from '../libraries/Locker.sol';
 
 contract LockAndMsgSender {
     error ContractLocked();
 
-    address internal constant NOT_LOCKED_FLAG = address(1);
-    address internal lockedBy = NOT_LOCKED_FLAG;
-
+    /// @notice Modifier enforcing a reentrancy lock that allows self-reentrancy
     modifier isNotLocked() {
+        // The contract is allowed to reenter itself to perform EXECUTE_SUB_PLAN commands
         if (msg.sender != address(this)) {
-            if (lockedBy != NOT_LOCKED_FLAG) revert ContractLocked();
-            lockedBy = msg.sender;
+            if (Locker.isLocked()) revert ContractLocked();
+            Locker.set(msg.sender);
             _;
-            lockedBy = NOT_LOCKED_FLAG;
+            Locker.set(address(0));
         } else {
             _;
         }
     }
 
-    /// @notice Calculates the recipient address for a command
-    /// @param recipient The recipient or recipient-flag for the command
-    /// @return output The resultant recipient for the command
-    function map(address recipient) internal view returns (address) {
-        if (recipient == Constants.MSG_SENDER) {
-            return lockedBy;
-        } else if (recipient == Constants.ADDRESS_THIS) {
-            return address(this);
-        } else {
-            return recipient;
-        }
+    /// @notice Function to be used instead of msg.sender, as the contract performs self-reentrancy and at
+    /// times msg.sender == address(this). Instead _msgSender() returns the initiator of the lock
+    function _msgSender() internal view returns (address) {
```

Comment:
I think this will have to override the _msgSender() in BaseActionsRouter..

Reply:
when you add the v4 swap lib

Reply:
yeah it will, or maybe theyll both have to be overriden by 1 implementing function. Not 100% sure yet until i pull in v4 and see what compiles LOL